delatore/sources/base.py

```diff
@@ -69,7 +70,9 @@ async def start(self, stop_event: asyncio.Event):
                 new = await asyncio.wait_for(self.get_update(), self.request_timeout)
             except (asyncio.TimeoutError, NoUpdates):
                 continue
-            if _same_status(new, last) and self.ignore_duplicates:
+            if self.ignore_duplicates and \
+                    same_status(new, last) and \
+                    _heartbeat_status(new, last) < self.heartbeat_interval:
```

Comment:
`_heartbeat_status(new, last) < self.heartbeat_interval` can be part of `same_status`

Reply:
no, it needs to be different checks